### ISSUE TITLE
Incorrect source location info in TUI mode

### ISSUE THREAD

**Issue by honggyukim** [namhyung/uftrace#416]
There are some cases that the source location is incorrectly shown in TUI mode.

Please download and test it with the below `uftrace.data.clang` file.

In the second session, `llvm::ScalarEvolution::isImpliedCond` is shown in `/mnt/data1/home3/honggyu/work/llvm/git/llvm/lib/IR/LegacyPassManager.cpp [line:673]` but actually it's incorrect.

It's actually pointing to the different location.
```
$ nm -l clang | c++filt | grep LegacyPassManager.cpp:673
0000000001011c50 T llvm::PMTopLevelManager::schedulePass(llvm::Pass*)   /mnt/data1/home3/honggyu/work/llvm/git/llvm/lib/IR/LegacyPassManager.cpp:673
```

I see that `.dbg` file doesn't have a problem, so I think it's incorrectly read in TUI mode.
```
        ...
F: 812010 llvm::ScalarEvolution::isImpliedCond
L: 9299 /mnt/data1/home3/honggyu/work/llvm/git/llvm/lib/Analysis/ScalarEvolution.cpp
F: 812350 llvm::ScalarEvolution::isImpliedCond
L: 9339 /mnt/data1/home3/honggyu/work/llvm/git/llvm/lib/Analysis/ScalarEvolution.cpp
        ...
```
`.dbg` information is same as the info from `nm -l` as follows:
```
$ nm -l clang | c++filt | grep -w isImpliedCond
0000000000c12010 T llvm::ScalarEvolution::isImpliedCond(llvm::CmpInst::Predicate, llvm::SCEV const*, llvm::SCEV const*, llvm::Value*, bool)       /mnt/data1/home3/honggyu/work/llvm/git/llvm/lib/Analysis/ScalarEvolution.cpp:9299
0000000000c12350 T llvm::ScalarEvolution::isImpliedCond(llvm::CmpInst::Predicate, llvm::SCEV const*, llvm::SCEV const*, llvm::CmpInst::Predicate, llvm::SCEV const*, llvm::SCEV const*)   /mnt/data1/home3/honggyu/work/llvm/git/llvm/lib/Analysis/ScalarEvolution.cpp:9339
```

**Comment by namhyung** [comment]
I found that the address is different between `nm` and `.dbg` - the offset is 0x400000 which is the loaded address of the executable.  I need to look at address space of clang built binary..

**Comment by namhyung** [comment]
Could you show me the output of `readelf -eW clang-7`?

**Comment by honggyukim** [comment]
It's kind of long but here it is.
```
$ readelf -eW clang-7                                                                                                                                                                                                                                                                                                 ELF Header:
  Magic:   7f 45 4c 46 02 01 01 03 00 00 00 00 00 00 00 00
  Class:                             ELF64
  Data:                              2's complement, little endian
  Version:                           1 (current)
  OS/ABI:                            UNIX - GNU
  ABI Version:                       0
  Type:                              EXEC (Executable file)
  Machine:                           Advanced Micro Devices X86-64
  Version:                           0x1
  Entry point address:               0x8eadd0
  Start of program headers:          64 (bytes into file)
  Start of section headers:          1714280728 (bytes into file)
  Flags:                             0x0
  Size of this header:               64 (bytes)
  Size of program headers:           56 (bytes)
  Number of program headers:         10
  Size of section headers:           64 (bytes)
  Number of section headers:         40
  Section header string table index: 37

Section Headers:
  [Nr] Name              Type            Address          Off    Size   ES Flg Lk Inf Al
  [ 0]                   NULL            0000000000000000 000000 000000 00      0   0  0
  [ 1] .interp           PROGBITS        0000000000400270 000270 00001c 00   A  0   0  1
  [ 2] .note.ABI-tag     NOTE            000000000040028c 00028c 000020 00   A  0   0  4
  [ 3] .note.gnu.build-id NOTE            00000000004002ac 0002ac 000024 00   A  0   0  4
  [ 4] .gnu.hash         GNU_HASH        00000000004002d0 0002d0 0329cc 00   A  5   0  8
  [ 5] .dynsym           DYNSYM          0000000000432ca0 032ca0 0cbca0 18   A  6   1  8
  [ 6] .dynstr           STRTAB          00000000004fe940 0fe940 296330 00   A  0   0  1
  [ 7] .gnu.version      VERSYM          0000000000794c70 394c70 010fb8 02   A  5   0  2
  [ 8] .gnu.version_r    VERNEED         00000000007a5c28 3a5c28 000280 00   A  6   9  8
  [ 9] .rela.dyn         RELA            00000000007a5ea8 3a5ea8 0071e8 18   A  5   0  8
  [10] .rela.plt         RELA            00000000007ad090 3ad090 001ab8 18  AI  5  26  8
  [11] .init             PROGBITS        00000000007aeb48 3aeb48 00001a 00  AX  0   0  4
  [12] .plt              PROGBITS        00000000007aeb70 3aeb70 0011e0 10  AX  0   0 16
  [13] .plt.got          PROGBITS        00000000007afd50 3afd50 000028 00  AX  0   0  8
  [14] .text             PROGBITS        00000000007afd80 3afd80 2cefe20 00  AX  0   0 16
  [15] .fini             PROGBITS        000000000349fba0 309fba0 000009 00  AX  0   0  4
  [16] .rodata           PROGBITS        000000000349fbc0 309fbc0 5cf0ec 00   A  0   0 32
  [17] .eh_frame_hdr     PROGBITS        0000000003a6ecac 366ecac 07ecbc 00   A  0   0  4
  [18] .eh_frame         PROGBITS        0000000003aed968 36ed968 2c1abc 00   A  0   0  8
  [19] .tbss             NOBITS          0000000003faf6a0 39af6a0 000008 00 WAT  0   0  8
  [20] .init_array       INIT_ARRAY      0000000003faf6a0 39af6a0 000850 00  WA  0   0  8
  [21] .fini_array       FINI_ARRAY      0000000003fafef0 39afef0 000008 00  WA  0   0  8
  [22] .jcr              PROGBITS        0000000003fafef8 39afef8 000008 00  WA  0   0  8
  [23] .data.rel.ro      PROGBITS        0000000003faff00 39aff00 24ecb8 00  WA  0   0 32
  [24] .dynamic          DYNAMIC         00000000041febb8 3bfebb8 000260 10  WA  6   0  8
  [25] .got              PROGBITS        00000000041fee18 3bfee18 0001e0 08  WA  0   0  8
  [26] .got.plt          PROGBITS        00000000041ff000 3bff000 000900 08  WA  0   0  8
  [27] .data             PROGBITS        00000000041ff900 3bff900 009740 00  WA  0   0 32
  [28] .bss              NOBITS          0000000004209040 3c09040 03a048 00  WA  0   0 32
  [29] .comment          PROGBITS        0000000000000000 3c09040 000033 01  MS  0   0  1
  [30] .debug_aranges    PROGBITS        0000000000000000 3c09073 15b880 00      0   0  1
  [31] .debug_info       PROGBITS        0000000000000000 3d648f3 49a57bca 00      0   0  1
  [32] .debug_abbrev     PROGBITS        0000000000000000 4d7bc4bd 79dc4f 00      0   0  1
  [33] .debug_line       PROGBITS        0000000000000000 4df5a10c 1833595 00      0   0  1
  [34] .debug_str        PROGBITS        0000000000000000 4f78d6a1 6035742 01  MS  0   0  1
  [35] .debug_loc        PROGBITS        0000000000000000 557c2de3 c8c7076 00      0   0  1
  [36] .debug_ranges     PROGBITS        0000000000000000 62089e59 35d0990 00      0   0  1
  [37] .shstrtab         STRTAB          0000000000000000 662dd79b 000178 00      0   0  1
  [38] .symtab           SYMTAB          0000000000000000 6565a7f0 24bec8 18     39 65561  8
  [39] .strtab           STRTAB          0000000000000000 658a66b8 a370e3 00      0   0  1
Key to Flags:
  W (write), A (alloc), X (execute), M (merge), S (strings), l (large)
  I (info), L (link order), G (group), T (TLS), E (exclude), x (unknown)
  O (extra OS processing required) o (OS specific), p (processor specific)

Program Headers:
  Type           Offset   VirtAddr           PhysAddr           FileSiz  MemSiz   Flg Align
  PHDR           0x000040 0x0000000000400040 0x0000000000400040 0x000230 0x000230 R E 0x8
  INTERP         0x000270 0x0000000000400270 0x0000000000400270 0x00001c 0x00001c R   0x1
      [Requesting program interpreter: /lib64/ld-linux-x86-64.so.2]
  LOAD           0x000000 0x0000000000400000 0x0000000000400000 0x39af424 0x39af424 R E 0x200000
  LOAD           0x39af6a0 0x0000000003faf6a0 0x0000000003faf6a0 0x2599a0 0x2939e8 RW  0x200000
  DYNAMIC        0x3bfebb8 0x00000000041febb8 0x00000000041febb8 0x000260 0x000260 RW  0x8
  NOTE           0x00028c 0x000000000040028c 0x000000000040028c 0x000044 0x000044 R   0x4
  TLS            0x39af6a0 0x0000000003faf6a0 0x0000000003faf6a0 0x000000 0x000008 R   0x8
  GNU_EH_FRAME   0x366ecac 0x0000000003a6ecac 0x0000000003a6ecac 0x07ecbc 0x07ecbc R   0x4
  GNU_STACK      0x000000 0x0000000000000000 0x0000000000000000 0x000000 0x000000 RW  0x10
  GNU_RELRO      0x39af6a0 0x0000000003faf6a0 0x0000000003faf6a0 0x24f960 0x24f960 R   0x1

 Section to Segment mapping:
  Segment Sections...
   00
   01     .interp
   02     .interp .note.ABI-tag .note.gnu.build-id .gnu.hash .dynsym .dynstr .gnu.version .gnu.version_r .rela.dyn .rela.plt .init .plt .plt.got .text .fini .rodata .eh_frame_hdr .eh_frame
   03     .init_array .fini_array .jcr .data.rel.ro .dynamic .got .got.plt .data .bss
   04     .dynamic
   05     .note.ABI-tag .note.gnu.build-id
   06     .tbss
   07     .eh_frame_hdr
   08
   09     .init_array .fini_array .jcr .data.rel.ro .dynamic .got
```

**Comment by namhyung** [comment]
I found a bug during symbol recording and pushed fix to [check/misc-fix](https://github.com/namhyung/uftrace/compare/check/misc-fix) branch.  Could you please test it again with newly recorded data?

**Comment by namhyung** [comment]
Fixed, thanks!

**Comment by namhyung** [comment]
I checked with the output `nm -l`, it has little differences - some filenames are missing and some line numbers are little bit different.  But there's no huge difference nor incorrect file names.  It's a quality issue but I think it's ok to go in for now.. ;-p

**Comment by honggyukim** [comment]
I also think that it's a quality issue, so maybe we can write a script to compare the result and then fix other problems later on. Thanks for the work!

**Comment by namhyung** [comment]
Merged: bee18bdf
